pandas/core/resample.py

```diff
@@ -1561,11 +1609,17 @@ def _get_period_bins(self, ax):
         end = ax.max().asfreq(self.freq, how="end")
         bin_shift = 0
 
-        # GH 23882
-        if self.base:
+        # GH 23882 & 31809
+        if self.origin is not None or self.offset is not None or self.base:
```

Comment:
can you simply remove all need for base (IOW convert it in the constructor)?

Reply:
Yep, it's done in the last commit!